webpack/JobInvocationDetail/JobInvocationToolbarButtons.js

```diff
@@ -0,0 +1,272 @@
+import PropTypes from 'prop-types';
+import React, { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import {
+  Button,
+  Dropdown,
+  DropdownItem,
+  DropdownPosition,
+  DropdownSeparator,
+  DropdownToggle,
+  Split,
+  SplitItem,
+} from '@patternfly/react-core';
+import { translate as __ } from 'foremanReact/common/I18n';
+import { foremanUrl } from 'foremanReact/common/helpers';
+import { STATUS as APIStatus } from 'foremanReact/constants';
+import { get } from 'foremanReact/redux/API';
+import {
+  cancelJob,
+  cancelRecurringLogic,
+  enableRecurringLogic,
+} from './JobInvocationActions';
+import {
+  STATUS,
+  GET_REPORT_TEMPLATES,
+  GET_REPORT_TEMPLATE_INPUTS,
+} from './JobInvocationConstants';
+import { selectTaskCancelable } from './JobInvocationSelectors';
+
+const JobInvocationToolbarButtons = ({
+  jobId,
+  data,
+  currentPermissions,
+  permissionsStatus,
+}) => {
+  const { succeeded, failed, task, recurrence, permissions } = data;
+  const recurringEnabled = recurrence?.state === 'active';
+  const canViewForemanTasks = permissions
+    ? permissions.view_foreman_tasks
+    : false;
+  const canEditRecurringLogic = permissions
+    ? permissions.edit_recurring_logics
+    : false;
+  const isTaskCancelable = useSelector(selectTaskCancelable);
+  const queryParams = new URLSearchParams({
+    [`report_template_report[input_values][${templateInputId}][value]`]: jobId,
+  });
+  const [isActionOpen, setIsActionOpen] = useState(false);
+  const [reportTemplateJobId, setReportTemplateJobId] = useState(undefined);
+  const [templateInputId, setTemplateInputId] = useState(undefined);
+
+  const dispatch = useDispatch();
+  // eslint-disable-next-line no-shadow
+  const onActionToggle = isActionOpen => {
+    setIsActionOpen(isActionOpen);
+  };
+  const onActionFocus = () => {
+    const element = document.getElementById(
+      'toggle-split-button-action-primary'
+    );
+    element.focus();
+  };
+  const onActionSelect = () => {
+    setIsActionOpen(false);
+    onActionFocus();
+  };
+  const hasPermission = permissionRequired =>
+    permissionsStatus === APIStatus.RESOLVED
+      ? currentPermissions?.some(
+          permission => permission.name === permissionRequired
+        )
+      : false;
+
+  useEffect(() => {
+    dispatch(
+      get({
+        key: GET_REPORT_TEMPLATES,
+        url: '/api/report_templates',
+        handleSuccess: ({ data: { results } }) => {
+          setReportTemplateJobId(
+            results.find(result => result.name === 'Job - Invocation Report')
```

Comment:
I guess it's the only way, hopefully it won't be renamed...

This is a third? time I see one is looking for a template by its name... I start to think we'd benefit from an additional metadata entry, something like `internal_id: :job_inv_rep`, which won't be ever changed. Not sure if it's as useful though... Maybe with an additional UI/API endpoint, which returns a template by that id with all the related information, such as inputs so we don't need to make 2 calls and search through all of them for 1 entry.

Reply:
Katello calls this label, it can be set on creation but it can never be modified. REX plugin uses concept of features, where one maps specific job template to a feature, which has hardcoded name. Third option that we use is through settings, e.g. Settings -> Ansible defines "Default Ansible inventory report template" where the user select specific report template. This even more common with provisioning templates. Each has pros and cons, but hopefully one of the existing mechanism could be used. If not and we'd introduce e.g. this internal_id metadata, we should consider the migration to it for other templates. Bonus points for further unification :-)

Reply:
Sorry, I wrote it in hurry without enough context. There are 3 options we already use in Foreman for such purpose.

1) some katello objects has `label` attribute used for this, which is close to the `internal_id` suggestion
2) remote execution plugin has Administer -> Remote Execution Features concept, which also has label which never changes. Each Feature is mapped to specific job template.
3) we use Settings for the template specification

I'd suggest (outside of this PR) to either adopt one of these three existing concepts, instead of hardcoding the template name in the code.

Reply:
Thank you all for your ideas, I'll incorporate them in some future PR. Besides that, everything should be fine (@adamruzicka).